### PR TITLE
Protect: Update use of 'vulnerability' naming to more generic 'threat' convention

### PR DESCRIPTION
**Pull request by nateweller**
This PR updates the terminology and naming conventions used in the Protect app, changing "vulnerabilities" to "threats". This change is made as part of the introduction of Scan API results in Protect data, which includes a more generic array of different threat types, not just vulnerabilities.

#### Changes proposed in this Pull Request:

* Updates the term "vulnerability" to "threat" in UI copy when describing Protect results. References specific to WPScan still use "vulnerability".
* Updates variable names using "vulnerability" to "threat" when related to Protect results. Variables derived from WPScan API still use "vulnerability".
* Note: naming of Tracks event names has been left unaffected for backwards compatibility.

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

1201069996155224-as-1202778527846723/f

p1660072159929849-slack-C029WFNV69M

p1HpG7-gSG-p2

#### Testing instructions:

* Boot up a fresh site and run through the entire Jetpack Protect process to view your first report, validating UI copy has been updated and there has been no regression in the application.
* Use the Jetpack Debug plugin to manually review each state of the application (loading, results with/without threats), validating UI copy has been updated and there has been no regression in the application.

Screen Shot 2022-08-19 at 5 17 47 AM

Screen Shot 2022-08-19 at 5 18 41 AM

### CODE REVIEW

File: projects/plugins/protect/src/js/hooks/use-protect-data/index.js [hunk 1]
```diff
@@ -22,10 +22,10 @@ export default function useProtectData() {
 	}
 
 	return {
-		numVulnerabilities: status.numVulnerabilities || 0,
-		numCoreVulnerabilities: status.core?.vulnerabilities?.length || 0,
-		numPluginsVulnerabilities: status.numPluginsVulnerabilities || 0,
-		numThemesVulnerabilities: status.numThemesVulnerabilities || 0,
+		numThreats: status.numThreats || 0,
+		numCoreThreats: status.core?.Threats?.length || 0,
```
Comment (dkmyta) [inline projects/plugins/protect/src/js/hooks/use-protect-data/index.js]: This was originally `status.core?.vulnerabilities?.length || 0`, should it now be `status.core?.threats?.length || 0` - no capitalization?

**Review by dkmyta — COMMENTED** [review]
I haven’t tested the display/functionality quite yet but in reviewing the changes to the code itself I found that the good majority of vul(s)/vuln(s)/vulnerability/vulnerabilities -> threat(s) changes look correct with only a single item that I was unclear on, I’ve added a comment directly to the line for further clarification.

File: projects/plugins/protect/src/js/hooks/use-protect-data/index.js (hunk 1 above)
Comment (nateweller) [reply projects/plugins/protect/src/js/hooks/use-protect-data/index.js]: Good catch @dkmyta! I've fixed this via [7cc5757](https://github.com/Automattic/jetpack/pull/25445/commits/7cc5757bab52eaebda9ecdd337455e8b89f71925) :+1:

**Comment by dkmyta** [comment]
This looks good to go - I reviewed the UI copy and all indicated locations and several others that were clearly describing Protect results (in all states) appear to be properly updated without any regression in the application. Please note that my review assumed that any other occurrences of the term "vulnerability" were intentionally left that way.

I do however see a number of failing checks that weren't failing previously and I don't believe the latest changes should have caused this. The associated errors do not appear to indicate any relationship either - perhaps forcing the checks to run again will provide the correct results.

**Review by dkmyta — APPROVED** [review]

**Pull request merged**